src/Routing/Route/DashedRoute.php

```diff
@@ -49,31 +50,47 @@ public function parse($url)
         if (!$params) {
             return false;
         }
-        if (!empty($params['controller'])) {
-            $params['controller'] = Inflector::camelize(str_replace(
-                '-',
-                '_',
-                $params['controller']
-            ));
-        }
         if (!empty($params['plugin'])) {
-            $params['plugin'] = Inflector::camelize(str_replace(
-                '-',
-                '_',
-                $params['plugin']
-            ));
+            $params['plugin'] = $this->_in($params['plugin']);
+        }
+        if (!empty($params['controller'])) {
+            $value = $params['controller'];
+            $params['controller'] = $this->_in($params['controller']);
+            if ($this->_out($params['controller']) !== $value) {
+                throw new NotFoundException();
+            }
         }
         if (!empty($params['action'])) {
-            $params['action'] = Inflector::variable(str_replace(
-                '-',
-                '_',
-                $params['action']
-            ));
+            $value = $params['action'];
+            $params['action'] = $this->_in($params['action']);
```

Comment:
Action name is supposed to be camelBacked not CamelCased which your new `_in()` method returns.